src/AbstractResourceListener.php

```diff
@@ -252,10 +253,10 @@ public function fetch($id)
     /**
      * Fetch all or a subset of resources
      *
-     * @param  array $params
+     * @param  Parameters $params
      * @return ApiProblem|mixed
      */
-    public function fetchAll($params = [])
+    public function fetchAll(Parameters $params)
```

Comment:
I think this needs to become `@param array|Parameters $params` for the sake of BC compatibility

Reply:
To be clear, you mean that we put both types array|Parameters in the docBlock and don't do a strong typing in the method definition as I suggested in the PR, correct?
And afterwards, we do the same change in the admin repository's template file?

Reply:
Correct, a native type addition here would be a BC break.

 > And afterwards, we do the same change in the admin repository's template file?

If the template is about code to be generated, and not code to be consumed, then a native type can be used there.

Reply:
done.

> If the template is about code to be generated, and not code to be consumed, then a native type can be used there.

The templates is used to generate the new resource class, but the `fetchAll()` inherits from the `AbstractResourceListener`. So, the method definition needs to stay the same I guess. I'll do a pull request in the admin repository updating the docblock in the same way.